frontend/src/components/App/AppContainer.tsx

```diff
@@ -31,6 +31,39 @@ window.desktopApi?.receive('backend-token', (token: string) => {
   setBackendToken(token);
 });
 
+/**
+ * QueryParamRedirect is a component that checks for a 'to' query parameter and redirects accordingly
+ * This should be placed near the top of your component hierarchy,
+ * typically in your main App component
+ * @returns
+ */
+const QueryParamRedirect = () => {
+  const history = useHistory();
+  const location = useLocation();
+
+  useEffect(() => {
+    // Get the current URL search params
+    const searchParams = new URLSearchParams(location.search);
+
+    // Check if 'to' parameter exists
+    const redirectPath = searchParams.get('to');
+
+    if (redirectPath) {
```

Comment:
This would be an "open redirect". If a malicious link like ?to=https://evil.example.com is opened, the app could redirect users off-site.

suggestion:
`if (redirectPath && redirectPath.startsWith('/')) {`

Reply:
Added a validation for the path